app/controllers/users/phone_setup_controller.rb

```diff
@@ -135,5 +139,31 @@ def new_phone_form_params
         :recaptcha_mock_score,
       )
     end
+
+    def check_phone_submission_limit(phone)
+      return false if phone.blank?
+      fingerprint = Pii::Fingerprinter.fingerprint(Phonelib.parse(phone).e164.to_s)
+      @submission_rate_limiter ||= RateLimiter.new(
+        target: fingerprint,
+        rate_limit_type: :phone_submissions,
+      )
+      @submission_rate_limiter.increment!
+      lock_out_user if @submission_rate_limiter.maxed?
+    end
+
+    def lock_out_user(user: current_user)
+      UpdateUser.new(user: user, attributes: { second_factor_locked_at: Time.zone.now }).call
+      sign_out
```

Comment:
As discussed in Slack, we should not log out the user when they become rate-limited. We should simply prevent them from continuing to submit the page.

Reply:
I have this working now, still using it's own RateLimiter. I'm trying to get a solution that works using the OtpRateLimiter sub. 
The current commit [0550b52](https://github.com/18F/identity-idp/pull/10459/commits/0550b52546c0cee7c4fc2372a653b150eceb48aa) doesn't log the user out anymore. But I noticed tracing back how the OtpRateLimiter is used here:
[if exceeded_otp_send_limit?](https://github.com/18F/identity-idp/blob/1d769f76fc396cfd2fca72a53b3cdbce956da5e1/app/controllers/users/two_factor_authentication_controller.rb#L178-L190)

it ends up here: (def handle_max_attempts)[https://github.com/18F/identity-idp/blob/1d769f76fc396cfd2fca72a53b3cdbce956da5e1/app/controllers/concerns/two_factor_authenticatable_methods.rb#L55-L80] and it looks like it signs the user out.